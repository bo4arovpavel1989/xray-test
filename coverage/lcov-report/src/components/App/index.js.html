<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/App/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/App</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>19/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>18/18</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import Header from '../Header'
import Menu from '../Menu'
import Test from '../Test'
import Admin from '../Admin'
import Forbidden from '../Forbidden'
import CreateTest from '../CreateTest'
import CreateQuestion from '../CreateQuestion'
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import '../../main.sass'
import { postData } from '../../actions.js'
import { setToken } from '../../helpers'
&nbsp;
class App extends React.Component {
  constructor (props) {
    super(props)
&nbsp;
    this.state = {
      isAdmin: false,
      authFail: false,
      isLogging: false,
      token: undefined
    }
&nbsp;
    this.login = this.login.bind(this)
    this.logoff = this.logoff.bind(this)
    this.handleAuth = this.handleAuth.bind(this)
&nbsp;
    this.postData = this.props.postData || <span class="branch-1 cbranch-no" title="branch not covered" >postData;</span>
    this.setToken = this.props.setToken || <span class="branch-1 cbranch-no" title="branch not covered" >setToken;</span>
  }
&nbsp;
  login (e) {
    e.preventDefault()
&nbsp;
    const data = {
      login: e.target.login.value,
      password: e.target.password.value
    }
&nbsp;
    this.setState({ isLogging: true })
&nbsp;
    return this.postData('login', data)
               .then(this.handleAuth)
               .catch(console.log)
               .finally(() =&gt; this.setState({ isLogging: false }))
  }
&nbsp;
  handleAuth (rep) {
    <span class="missing-if-branch" title="else path not taken" >E</span>if (rep.auth) this.setToken(rep.token);
    this.setState({ isAdmin: rep.auth, authFail: !rep.auth, token: rep.token })
  }
&nbsp;
  logoff () {
    this.setState({
      isAdmin: false,
      token: undefined
     })
&nbsp;
     this.setToken(null)
  }
&nbsp;
  render () {
    const { isAdmin, authFail, isLogging } = this.state
&nbsp;
    return (
      &lt;BrowserRouter&gt;
        &lt;div&gt;
          &lt;Header
           isAdmin={isAdmin}
           authFail={authFail}
           isLogging = {isLogging}
           login={this.login}
           logoff={this.logoff}
           /&gt;
          &lt;Switch&gt;
            &lt;Route path='/' render={Menu} exact/&gt;
            &lt;Route path='/test' render={Test}/&gt;
            &lt;Route path='/admin' render={isAdmin ? Admin : Forbidden}/&gt;
            &lt;Route path='/create/test' render={isAdmin ? CreateTest : Forbidden}/&gt;
            &lt;Route path='/create/question' render={isAdmin ? CreateQuestion : Forbidden}/&gt;
          &lt;/Switch&gt;
        &lt;/div&gt;
      &lt;/BrowserRouter&gt;
    )
  }
}
&nbsp;
export default App;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Apr 25 2019 09:09:35 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
