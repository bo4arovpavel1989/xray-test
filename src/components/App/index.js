import React from 'react'
import Header from '../Header'
import Menu from '../Menu'
import Test from '../Test'
import Admin from '../Admin'
import Forbidden from '../Forbidden'
import CreateTest from '../CreateTest'
import CreateQuestion from '../CreateQuestion'
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import '../../main.sass'
import { postData } from '../../actions.js'
import { setToken } from '../../helpers'

class App extends React.Component {
  constructor () {
    super()

    this.state = {
      isAdmin: false,
      authFail: false,
      isLogging: false,
      token: undefined
    }

    this.login = this.login.bind(this)
    this.logoff = this.logoff.bind(this)
    this.handleAuth = this.handleAuth.bind(this)
  }

  login (e) {
    e.preventDefault()

    const data = {
      login: e.target.login.value,
      password: e.target.password.value
    }

    this.setState({ isLogging: true })

    postData('login', data)
      .then(this.handleAuth)
      .catch(console.log)
      .finally(() => this.setState({ isLogging: false }))
  }

  handleAuth (rep) {
    if (rep.auth) setToken(rep.token)
    this.setState({ isAdmin: rep.auth, authFail: !rep.auth, token: rep.token })
  }

  logoff () {
    this.setState({
      isAdmin: false,
      token: undefined
     })

     setToken(null)
  }

  render () {
    const { isAdmin, authFail, isLogging } = this.state

    return (
      <BrowserRouter>
        <div>
          <Header
           isAdmin={isAdmin}
           authFail={authFail}
           isLogging = {isLogging}
           login={this.login}
           logoff={this.logoff}
           />
          <Switch>
            <Route path='/' render={Menu} exact/>
            <Route path='/test' render={Test}/>
            <Route path='/admin' render={isAdmin ? Admin : Forbidden}/>
            <Route path='/create/test' render={isAdmin ? CreateTest : Forbidden}/>
            <Route path='/create/question' render={isAdmin ? CreateQuestion : Forbidden}/>
          </Switch>
        </div>
      </BrowserRouter>
    )
  }
}

export default App;
