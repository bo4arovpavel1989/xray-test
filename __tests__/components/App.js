import React from 'react';
import App from '../../src/components/App';
import Enzyme, { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {
  const props = {
    setToken: jest.fn(),
    postData: jest.fn(() => new Promise((res, rej) => res({ auth: true, token: 'testToken' })))
  };
  const initialState = {
    isAdmin: false,
    authFail: false,
    isLogging: false,
    token: undefined
  }
  const app = mount(<App { ...props }/>);

  it('should render correct', () => {
      expect(shallowToJson(app)).toMatchSnapshot()
  });

  describe('should handle login', () => {
    const e = {
      preventDefault: jest.fn(),
      target: {
        login: { value: 'testLogin' },
        password: { value: 'testPassword' }
      }
    };
    const data = {
      login: e.target.login.value,
      password: e.target.password.value
    };

    beforeAll(() => {
      return app.instance().login(e);
    })

    afterAll(() => {
      app.instance().setState(initialState);
      props.postData.mockClear();
      props.setToken.mockClear();
    })

    it('should call preventDefault', () => {
      expect(e.preventDefault).toHaveBeenCalledTimes(1);
    })

    it('should call postData', () => {
      expect(props.postData).toHaveBeenCalledWith('login', data);
    })

    it('should call setToken', () => {
      expect(props.setToken).toHaveBeenCalledTimes(1);
    })

    it('should change state', () => {
      expect(app.instance().state).toEqual({
        isAdmin: true,
        authFail: false,
        token: 'testToken',
        isLogging: false
      });
    })
  })
})
