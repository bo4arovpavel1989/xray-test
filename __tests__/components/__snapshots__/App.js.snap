// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correct 1`] = `
<App
  postData={[MockFunction]}
  setToken={[MockFunction]}
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <withRouter(Header)
          authFail={false}
          isAdmin={false}
          isLogging={false}
          login={[Function]}
          logoff={[Function]}
        >
          <Route>
            <Header
              authFail={false}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              isAdmin={false}
              isLogging={false}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              login={[Function]}
              logoff={[Function]}
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div
                className="headerArea"
              >
                <Link
                  className="header"
                  replace={false}
                  to="/"
                >
                  <a
                    className="header"
                    href="/"
                    onClick={[Function]}
                  >
                    <h1>
                      XRay-test
                    </h1>
                  </a>
                </Link>
                <withRouter(Login)
                  authFail={false}
                  login={[Function]}
                >
                  <Route>
                    <Login
                      authFail={false}
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      login={[Function]}
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                    >
                      <form
                        className="loginForm"
                        onSubmit={[Function]}
                      >
                        <span>
                          <input
                            name="login"
                            placeholder="login"
                            type="login"
                          />
                          <input
                            name="password"
                            placeholder="password"
                            type="password"
                          />
                        </span>
                        <span>
                          <input
                            type="submit"
                            value="Войти"
                          />
                        </span>
                      </form>
                    </Login>
                  </Route>
                </withRouter(Login)>
              </div>
            </Header>
          </Route>
        </withRouter(Header)>
        <Switch>
          <Route
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
            render={[Function]}
          >
            <Route>
              <Menu
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <div
                  className="container"
                >
                  <div
                    className="menuItem"
                  >
                    <Link
                      className="menuButton"
                      replace={false}
                      to="/test"
                    >
                      <a
                        className="menuButton"
                        href="/test"
                        onClick={[Function]}
                      >
                        Начать тест
                      </a>
                    </Link>
                  </div>
                  <div
                    className="menuItem"
                  >
                    <Link
                      className="menuButton"
                      replace={false}
                      to="/admin"
                    >
                      <a
                        className="menuButton"
                        href="/admin"
                        onClick={[Function]}
                      >
                        Настройки
                      </a>
                    </Link>
                  </div>
                </div>
              </Menu>
            </Route>
          </Route>
        </Switch>
      </div>
    </Router>
  </BrowserRouter>
</App>
`;
